### PR TITLE
Storage should work even if no metadata.json exists.

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

**Comment by Xazax-hun** [comment]
I think it should be easy to test this. You could delete the metadata json after the analyze and try store again.

**Comment by whisperity** [comment]
Well, I mean, it's okay that this is easy to test, but for resillience, I'd rather add a test case for it directly into the code.

**Comment by Xazax-hun** [comment]
I agree with @whisperity

File: tests/functional/detection_status/test_detection_status.py [hunk 1]
```diff
@@ -100,9 +100,21 @@ def _create_source_file(self, version):
         with open(os.path.join(self._test_dir, self._source_file), 'w') as f:
             f.write(self.sources[version])
 
-        codechecker.check(self._codechecker_cfg,
+        codechecker.analyze(self._codechecker_cfg,
+                            'hello',
+                            self._test_dir)
+
+        try:
```
Comment (gyorb) [inline tests/functional/detection_status/test_detection_status.py]: Couldn't we create a separate test for the metadata.json removal? At least a separate test case if not a separate test package.

**Pull request merged**